### PR TITLE
SED-1513 created a dialog to display STEP versions

### PR DESCRIPTION
**Pull request by bojidaryovchev**

### CODE REVIEW

**Comment by dvladir** [comment]
@bojidaryovchev As other app's modals have buttons in the footer, I think it would be correct to add a close button to this new modal too.

**Comment by dvladir** [comment]
@neogucky 
I am not sure the version's replacement works properly after it was moved from `index.html` to sidebar component and login page. On the deployed versions, I see the placeholders in the html layout.

**Review by dvladir — CHANGES_REQUESTED** [review]
@bojidaryovchev Add the button to modal

@neogucky If version's replacement during the pipeline doesn't work properly, we need to fix it first, before merging this PR

**Comment by neogucky** [comment]
> @neogucky I am not sure the version's replacement works properly after it was moved from `index.html` to sidebar component and login page. On the deployed versions, I see the placeholders in the html layout.

Actually we have some problem with the placeholders at the moment. In theory they should be replaced in all html and js files

**Review by neogucky — CHANGES_REQUESTED** [review]
Looks very good, thank you! You couldn't know this but the version placeholders are replaced over all htmkl files in our build version, so you can just add them directly into the modal instad of adding them as parameters which makes the modal more streight forward.

File: projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <strong class="flex-1">Version</strong>
+      </div>
+
+      <div class="list-item">
+        <div class="flex-1">STEP</div>
+        <div class="flex-1">
+          <span style="display: none" id="frontend-os">{{ '${project.versionFEOS}' }}</span>
+          <span style="display: none" id="frontend-ee">{{ '${project.versionFEEE}' }}</span>
+        </div>
+      </div>
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html:17]:
```suggestion
      <div class="list-item" id="frontend-os">
        <div class="flex-1">Step base</div>
        <div class="flex-1">
          <span style="display: none">{{ '${project.versionFEOS}' }}</span>
        </div>
      </div>
    <div class="list-item"  id="frontend-ee">
        <div class="flex-1">Step enterprise</div>
        <div class="flex-1">
          <span style="display: none">{{ '${project.versionFEEE}' }}</span>
        </div>
      </div>
```

Comment (neogucky) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html:17]: Please check code if you accept the suggestion, I just wrote it untested. The idea would be that we have two Step versions in Step EE and it is super helpful for us to know which ones are deployed. Makes it more clear to have them in their own row

File: projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    {
+      id: VersionDialogTabs.BE,
+      label: 'BackEnd',
+    },
+  ];
+  readonly VersionDialogTabs = VersionDialogTabs;
+
+  versionFEOS?: string;
+  versionFEEE?: string;
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.ts:29]: These aren't used anymore, correct?

File: projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html (hunk 1 above)
Comment (neogucky) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html:17]: And I think currently the version is not hidden when ${project.versionFEEE} is not replaced, correct?

**Review by neogucky — CHANGES_REQUESTED** [review]
@bojidaryovchev Thanks, I left some comments! Can you also check if we can use this with the modal component?

File: projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.ts (hunk 2 above)
Comment (bojidaryovchev) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.ts:29]: nice catch

File: projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html (hunk 1 above)
Comment (bojidaryovchev) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html:17]: well I am not really sure, I used the snipped you've suggested and this is what I see 
image

Comment (bojidaryovchev) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html:17]: however I am not really sure about those ids being attached to the list-item divs, are you sure this is how it should be? I can't really make an informed decision here

Comment (bojidaryovchev) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html:17]: image

Comment (neogucky) [reply projects/step-frontend/src/lib/modules/_common/components/versions-dialog/versions-dialog.component.html]: You are right, we need to also pull up the display:none into the list-item level. And we also need the logic to hide them when the placeholder is not replaced or empty. I will check tomorrow with DevOps so we can hopefully test it.

**Review by neogucky — APPROVED** [review]

**Pull request merged**